pkg/zcapld/create.go

```diff
@@ -0,0 +1,240 @@
+func WithID(id string) CapabilityOption {
+	return func(o *CapabilityOptions) {
+		o.ID = id
+	}
+}
+
+// WithParent identifies the parent Capability.
+func WithParent(p string) CapabilityOption {
+	return func(o *CapabilityOptions) {
+		o.Parent = p
+	}
+}
+
+// WithInvoker identifies the invoker of the Capability.
+func WithInvoker(i string) CapabilityOption {
+	return func(o *CapabilityOptions) {
+		o.Invoker = i
+	}
+}
+
+// WithController identifies the controller of the Capability.
+func WithController(c string) CapabilityOption {
+	return func(o *CapabilityOptions) {
+		o.Controller = c
+	}
+}
+
+// WithDelegator identifies the delegator of the Capability.
+func WithDelegator(d string) CapabilityOption {
+	return func(o *CapabilityOptions) {
+		o.Delegator = d
+	}
+}
+
+// WithAllowedActions sets the actions allowed by the Capability.
+func WithAllowedActions(actions ...string) CapabilityOption {
+	return func(o *CapabilityOptions) {
+		o.AllowedAction = actions
+	}
+}
+
+// WithInvocationTarget sets the invocation target on the Capability.
+func WithInvocationTarget(targetID, targetType string) CapabilityOption {
+	return func(o *CapabilityOptions) {
+		o.InvocationTarget = InvocationTarget{
+			ID:   targetID,
+			Type: targetType,
+		}
+	}
+}
+
+// WithChallenge sets the challenge to include in the proof.
+func WithChallenge(c string) CapabilityOption {
+	return func(o *CapabilityOptions) {
+		o.Challenge = c
+	}
+}
+
+// WithDomain sets the domain to include in the proof.
+func WithDomain(d string) CapabilityOption {
+	return func(o *CapabilityOptions) {
+		o.Domain = d
+	}
+}
+
+// WithCapabilityChain specifies the capabilityChain on the proof for the Capability.
+func WithCapabilityChain(chain ...interface{}) CapabilityOption {
+	return func(o *CapabilityOptions) {
+		o.CapabilityChain = chain
+	}
+}
+
+// Signer signs the Capability.
+type Signer struct {
+	ariessigner.SignatureSuite
+	SuiteType          string
+	VerificationMethod string
+}
+
+// NewCapability constructs a new, signed Capability with the options provided.
+func NewCapability(signer *Signer, options ...CapabilityOption) (*Capability, error) {
+	if signer == nil {
+		return nil, errors.New("must provide a signer")
+	}
+
+	opts := &CapabilityOptions{
+		ID: fmt.Sprintf("urn:uuid:%s", uuid.New().String()),
+	}
+
+	for i := range options {
+		options[i](opts)
+	}
+
+	zcap := &Capability{
+		Context:          SecurityContextV2,
+		ID:               opts.ID,
+		Invoker:          opts.Invoker,
+		Controller:       opts.Controller,
+		Delegator:        opts.Delegator,
+		Parent:           opts.Parent,
+		AllowedAction:    opts.AllowedAction,
+		InvocationTarget: opts.InvocationTarget,
+	}
+
+	err := signZCAP(zcap, signer, opts)
+	if err != nil {
+		return nil, fmt.Errorf("failed to sign zcap: %w", err)
+	}
+
+	return zcap, nil
+}
+
+func signZCAP(zcap *Capability, sig *Signer, options *CapabilityOptions) error {
```

Comment:
Note that `sig` is commonly used as an abbreviated version of "signature". Consider renaming to `sign` or just `signer`.

Reply:
renamed to `signer`